cos-fleetshard-operator/src/main/java/org/bf2/cos/fleetshard/operator/connector/ConnectorConfiguration.java

```diff
@@ -64,9 +64,9 @@ public ConnectorConfiguration(ObjectNode connectorSpec, Class<S> type) {
 
         connectorSpec.remove(RESERVED_PROPERTIES);
 
-        if (type.isAssignableFrom(String.class)) {
+        if (connectorSpecType.isAssignableFrom(String.class)) {
             this.connectorSpec = (S) connectorSpec.toString();
-        } else if (type.isAssignableFrom(Properties.class)) {
+        } else if (connectorSpecType.isAssignableFrom(Properties.class)) {
             Properties result = new Properties();
             try {
                 result.load(new StringReader(connectorSpec.toString()));
```

Comment:
is this even supposed to work given that as connector Spec is an ObjectNode

Reply:
That is code that was previously used for the connectorSpec type in `org.bf2.cos.fleetshard.support.resources.Secrets#extract(..)`. I just copied it here as Deserialization for this moved here for the properties that were left. Not sure if we need it. I just copied it with the previous-previous PR.

Reply:
because in a secret you may have a string used to represent an application.properties  i.e.

```yaml
data:
    application.properties: ${base64}
```

So that method was an helper to extract properties from a secret data entry

Reply:
ah. so this is not the case for connectorSpec, true! I will drop the code then